### PR TITLE
Ref #17769 - Replace superglobals with ServerRequest in Import & Export Controllers

### PR DESCRIPTION
**Pull request by ru-asdx**
Signed-off-by: Evgeny Skorlov <eugene@skorlov.name>

### Description

Ref https://github.com/phpmyadmin/phpmyadmin/issues/17769

Replace superglobals with ServerRequest in Import & Export Controllers

### CODE REVIEW

File: libraries/classes/Controllers/Export/TablesController.php [hunk 1]
```diff
@@ -25,7 +25,7 @@ public function __construct(ResponseRenderer $response, Template $template, Expo
 
     public function __invoke(ServerRequest $request): void
     {
-        if (empty($_POST['selected_tbl'])) {
+        if ($request->getParsedBodyParam('selected_tbl') === null) {
```
Comment (kamil-tekiela) [inline libraries/classes/Controllers/Export/TablesController.php:28]: Can it not be an empty string too? If not then using `hasBodyParam` would be more suitable.

File: libraries/classes/Controllers/Import/ImportController.php [hunk 2]
```diff
@@ -138,9 +138,9 @@ public function __invoke(ServerRequest $request): void
         // (eg. non import, but query box/window run)
         if (! empty($GLOBALS['sql_query'])) {
             // apply values for parameters
-            if (! empty($_POST['parameterized']) && ! empty($_POST['parameters']) && is_array($_POST['parameters'])) {
-                /** @var array<string, string> $parameters */
-                $parameters = $_POST['parameters'];
+            /** @var array<string, string> $parameters */
+            $parameters = $request->getParsedBodyParam('parameters');
+            if ($request->hasBodyParam('parameterized') && $parameters !== null && is_array($parameters)) {
```
Comment (kamil-tekiela) [inline libraries/classes/Controllers/Import/ImportController.php:143]: This is redundant. Either it's an array or null. It cannot be both.

File: libraries/classes/Controllers/Import/ImportController.php [hunk 3]
```diff
@@ -169,7 +169,7 @@ public function __invoke(ServerRequest $request): void
             $_SESSION['sql_from_query_box'] = true;
 
             // If there is a request to ROLLBACK when finished.
-            if (isset($_POST['rollback_query'])) {
+            if ($request->getParsedBodyParam('rollback_query') !== null) {
```
Comment (kamil-tekiela) [inline libraries/classes/Controllers/Import/ImportController.php:172]:
```suggestion
            if ($request->hasBodyParam('rollback_query')) {
```

File: libraries/classes/Controllers/Import/ImportController.php [hunk 4]
```diff
@@ -204,15 +204,17 @@ public function __invoke(ServerRequest $request): void
...
             $GLOBALS['import_type'] = 'query';
             $GLOBALS['format'] = 'sql';
         }
 
         // If we didn't get any parameters, either user called this directly, or
         // upload limit has been reached, let's assume the second possibility.
-        if ($_POST == [] && $_GET == []) {
+        $getParams = $request->getQueryParams();
+        $postParams = $request->getParsedBody();
+        if (is_array($postParams) && $postParams == [] && is_array($getParams) && $getParams == []) {
```
Comment (kamil-tekiela) [inline libraries/classes/Controllers/Import/ImportController.php:217]:
```suggestion
        if ($postParams === [] && $getParams === []) {
```

File: libraries/classes/Controllers/Import/ImportController.php [hunk 5]
```diff
@@ -326,24 +329,27 @@ public function __invoke(ServerRequest $request): void
         $GLOBALS['result'] = false;
 
         // Bookmark Support: get a query back from bookmark if required
-        if (! empty($_POST['id_bookmark'])) {
-            $id_bookmark = (int) $_POST['id_bookmark'];
-            switch ($_POST['action_bookmark']) {
+        $id_bookmark = $request->getParsedBodyParam('id_bookmark');
+        if (! empty($id_bookmark)) {
```
Comment (kamil-tekiela) [inline libraries/classes/Controllers/Import/ImportController.php:333]: Can you make this check more specific, please? Maybe move the cast above the if and then check against 0?

File: libraries/classes/Controllers/Import/ImportController.php [hunk 6]
```diff
@@ -326,24 +329,27 @@ public function __invoke(ServerRequest $request): void
...
+            switch ($action_bookmark) {
                 case 0: // bookmarked query that have to be run
                     $bookmark = Bookmark::get(
                         $this->dbi,
                         $GLOBALS['cfg']['Server']['user'],
                         DatabaseName::fromValue($GLOBALS['db']),
                         $id_bookmark,
                         'id',
-                        isset($_POST['action_bookmark_all'])
+                        $request->getParsedBodyParam('action_bookmark_all') !== null
```
Comment (kamil-tekiela) [inline libraries/classes/Controllers/Import/ImportController.php:344]:
```suggestion
                        $request->hasBodyParam('action_bookmark_all')
```

File: libraries/classes/Controllers/Import/ImportController.php [hunk 7]
```diff
@@ -326,24 +329,27 @@ public function __invoke(ServerRequest $request): void
...
                     );
                     if (! $bookmark instanceof Bookmark) {
                         break;
                     }
 
-                    if (! empty($_POST['bookmark_variable'])) {
-                        $GLOBALS['import_text'] = $bookmark->applyVariables($_POST['bookmark_variable']);
+                    $bookmark_variable = $request->getParsedBodyParam('bookmark_variable');
+                    if (! empty($bookmark_variable)) {
```
Comment (kamil-tekiela) [inline libraries/classes/Controllers/Import/ImportController.php:351]: Please make this more specific too.

File: libraries/classes/Controllers/Import/ImportController.php [hunk 8]
```diff
@@ -550,8 +556,8 @@ public function __invoke(ServerRequest $request): void
         }
 
         // Something to skip? (because timeout has passed)
-        if (! $GLOBALS['error'] && isset($_POST['skip'])) {
-            $original_skip = $skip = intval($_POST['skip']);
+        if (! $GLOBALS['error'] && $request->getParsedBodyParam('skip') !== null) {
```
Comment (kamil-tekiela) [inline libraries/classes/Controllers/Import/ImportController.php:559]:
```suggestion
        if (! $GLOBALS['error'] && $request->hasBodyParam('skip')) {
```

File: libraries/classes/Controllers/Import/ImportController.php [hunk 9]
```diff
@@ -611,11 +617,11 @@ public function __invoke(ServerRequest $request): void
         }
 
         // Show correct message
-        if (! empty($id_bookmark) && $_POST['action_bookmark'] == 2) {
+        if (! empty($id_bookmark) && $action_bookmark == 2) {
```
Comment (kamil-tekiela) [inline libraries/classes/Controllers/Import/ImportController.php:620]: I don't think `$action_bookmark` is defined in this scope.

File: libraries/classes/Controllers/Import/ImportController.php [hunk 10]
```diff
@@ -611,11 +617,11 @@ public function __invoke(ServerRequest $request): void
...
             $GLOBALS['message'] = Message::success(__('The bookmark has been deleted.'));
             $GLOBALS['display_query'] = $GLOBALS['import_text'];
             $GLOBALS['error'] = false; // unset error marker, it was used just to skip processing
-        } elseif (! empty($id_bookmark) && $_POST['action_bookmark'] == 1) {
+        } elseif (! empty($id_bookmark) && $action_bookmark == 1) {
```
Comment (kamil-tekiela) [inline libraries/classes/Controllers/Import/ImportController.php:624]: What type is `$id_bookmark`? Please make the check more specific.

File: libraries/classes/Controllers/Import/ImportController.php [hunk 11]
```diff
@@ -773,16 +779,16 @@ public function __invoke(ServerRequest $request): void
...
                 $this->sql->storeTheQueryAsBookmark(
                     $relation->getRelationParameters()->bookmarkFeature,
                     $GLOBALS['db'],
                     $GLOBALS['cfg']['Server']['user'],
-                    $_POST['sql_query'],
-                    $_POST['bkm_label'],
-                    isset($_POST['bkm_replace'])
+                    $request->getParsedBodyParam('sql_query'),
+                    $request->getParsedBodyParam('bkm_label'),
+                    $request->getParsedBodyParam('bkm_replace') !== null
```
Comment (kamil-tekiela) [inline libraries/classes/Controllers/Import/ImportController.php:791]:
```suggestion
                    $request->hasBodyParam('bkm_replace') 
```

File: libraries/classes/Controllers/Import/ImportController.php [hunk 12]
```diff
@@ -794,16 +800,16 @@ public function __invoke(ServerRequest $request): void
...
```
Comment (kamil-tekiela) [inline libraries/classes/Controllers/Import/ImportController.php:812]:
```suggestion
                    $request->hasBodyParam('bkm_replace')
```

File: libraries/classes/Controllers/Import/ImportController.php [hunk 13]
```diff
@@ -823,7 +829,7 @@ public function __invoke(ServerRequest $request): void
         }
 
         // If there is request for ROLLBACK in the end.
-        if (! isset($_POST['rollback_query'])) {
+        if ($request->getParsedBodyParam('rollback_query') === null) {
```
Comment (kamil-tekiela) [inline libraries/classes/Controllers/Import/ImportController.php:832]:
```suggestion
        if (! $request->getParsedBodyParam('rollback_query')) {
```

File: libraries/classes/Controllers/Import/StatusController.php [hunk 14]
```diff
@@ -46,7 +46,7 @@ public function __invoke(ServerRequest $request): void
         ] = Ajax::uploadProgressSetup();
 
         // $_GET["message"] is used for asking for an import message
-        if (isset($_GET['message']) && $_GET['message']) {
+        if ($request->getQueryParam('message') !== null) {
```
Comment (kamil-tekiela) [inline libraries/classes/Controllers/Import/StatusController.php:49]: Are you sure it cannot be anything else then null? Why not use `hasQueryParam` instead?

**Review by kamil-tekiela — CHANGES_REQUESTED** [review]

**Review by MauricioFauth — APPROVED** [review]

**Pull request merged**